### DISCUSSION TITLE
Segmentation fault, mac os 13.

### DISCUSSION THREAD

**Discussion by arogozhnikov** [psycopg/psycopg2#1538] [General]
Don't want this to be an issue, since it likely isn't.

I'm using sqlalchemy + psycopg2. 

Exact script I'm talking about is single-threaded insertion of data.

It worked well for a while, and at some point it just started crashing with segfault when session for sqlalchemy is closed.
(I.e. problems are not during commit, but rather during closing connection).

I believe the reason is my update to mac os 13 (M1 processor) that happened around that time.
Reinstalling psycopg2-binary or psycopg2 and updating to newest version did not help.

Then I reinstalled the python from brew and reinstalled venv - and then the problem disappeared.

Just sharing if someone hits the same problem.

**Comment by br34th5** [comment]
Installing psycopg2 for the first time on my Linux Mint 21.3 Cinnamon environment, I was also getting various errors:
- free(): invalid pointer
- munmap_chunk(): invalid pointer
- 23476 segmentation fault (core dumped)
then followed your steps (made new venv with reinstalled python) - no more errors.
